File: scr/com/itkacher/views/form/DataForm.java
package com.itkacher.views.form;

import com.intellij.ui.JBSplitter;
import com.intellij.ui.components.JBPanel;
import com.intellij.ui.components.JBScrollPane;
import com.intellij.ui.components.JBTabbedPane;
import com.intellij.ui.components.JBTextField;
import com.intellij.ui.table.JBTable;
import com.intellij.uiDesigner.core.GridConstraints;

import javax.swing.*;
import java.awt.*;

public class DataForm {
    private JBPanel dataPanel;
    private JBTable requestTable1;
    private JBTabbedPane tabsPane1;


    public JPanel getDataPanel() {
        return dataPanel;
    }

    public JTable getRequestTable() {
        return requestTable1;
    }

    public JTabbedPane getTabsPane() {
        return tabsPane1;
    }


    public DataForm() {
        dataPanel = new JBPanel(new BorderLayout());
        JBPanel panel1 = new JBPanel(new BorderLayout());
        JBPanel panel2 = new JBPanel(new BorderLayout());

        requestTable1 = new JBTable();
        tabsPane1 = new JBTabbedPane();

        JBScrollPane scroll1 = new JBScrollPane(requestTable1);

        panel1.add(scroll1, BorderLayout.CENTER);
        panel2.add(tabsPane1, BorderLayout.CENTER);

        JBSplitter splitter = new JBSplitter(true);
        splitter.setProportion(0.5f);
        splitter.setFirstComponent(panel1);
        splitter.setSecondComponent(panel2);

        dataPanel.removeAll();
        dataPanel.add(splitter, BorderLayout.CENTER);
    }

}

File: src/main/kotlin/com/itkacher/views/form/DataForm.kt
/**
 * Copyright 2018 LocaleBro.com [Ievgenii Tkachenko(gektor650@gmail.com)]
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.itkacher.views.form

import com.intellij.ui.JBSplitter
import com.intellij.ui.components.JBTabbedPane
import com.intellij.ui.table.JBTable
import com.itkacher.views.FrameScrollPanel
import com.itkacher.views.SimpleJBPanel
import java.awt.BorderLayout

class DataForm {
    val dataPanel = SimpleJBPanel()
    val requestTable = JBTable()
    val tabsPane = JBTabbedPane()


    init {
        val panel1 = FrameScrollPanel(requestTable)
        val panel2 = SimpleJBPanel()
        panel2.add(tabsPane, BorderLayout.CENTER)
        val splitter = JBSplitter(true)
        splitter.proportion = 0.5f
        splitter.firstComponent = panel1
        splitter.secondComponent = panel2
        dataPanel.add(splitter)
    }

}
